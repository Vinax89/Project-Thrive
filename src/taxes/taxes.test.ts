import { describe, it, expect } from 'vitest';
import { estimateTax, type FilingStatus } from './index';

describe('estimateTax', () => {
  const boundaries: Record<FilingStatus, number[]> = {
    single: [11600, 47150, 100525, 191950, 243725, 609350],
    married_joint: [23200, 94300, 201050, 383900, 487450, 731200],
    married_separate: [11600, 47150, 100525, 191950, 243725, 365600],
    head: [16550, 63100, 100500, 191950, 243700, 609350]
  };

  it('calculates federal tax at bracket boundaries for all filing statuses', () => {
    const results: Record<FilingStatus, number[]> = {
      single: [],
      married_joint: [],
      married_separate: [],
      head: []
    };
    for (const status of Object.keys(boundaries) as FilingStatus[]) {
      results[status] = boundaries[status].map((income) =>
        estimateTax('federal', 2025, income, status)
      );
    }
    expect(results).toMatchSnapshot();
  });

  it('returns 0 for no income across states', () => {
    for (const state of ['federal', 'california', 'oregon']) {
      expect(estimateTax(state, 2025, 0)).toBe(0);
    }
  });

  it('calculates tax for multiple states and years', () => {
    expect(estimateTax('federal', 2025, 1_000_000)).toBe(328187.75);
    expect(estimateTax('california', 2025, 1_000_000)).toBe(104989.14);
    expect(estimateTax('oregon', 2025, 1_000_000)).toBe(97254.5);
    // 2024 uses same tables in tests
    expect(estimateTax('federal', 2024, 1_000_000)).toBe(328187.75);
  });
});
