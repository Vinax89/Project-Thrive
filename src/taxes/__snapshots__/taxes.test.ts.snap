// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`estimateTax > calculates federal tax at bracket boundaries for all filing statuses 1`] = `
{
  "head": [
    1655,
    7241,
    15469,
    37417,
    53977,
    181954.5,
  ],
  "married_joint": [
    2320,
    10852,
    34337,
    78221,
    111357,
    196669.5,
  ],
  "married_separate": [
    1160,
    5426,
    17168.5,
    39110.5,
    55678.5,
    98334.75,
  ],
  "single": [
    1160,
    5426,
    17168.5,
    39110.5,
    55678.5,
    183647.25,
  ],
}
`;
